`type annotations needed` when building with `log` crate

### Polars version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://crates.io/crates/polars) of Polars.


### Issue description

polars failed to build when theres a dependency on [rust-lang/log](https://github.com/rust-lang/log) + `features = ["kv_unstable_serde"]`  
tested on version `0.27.2` and latest master  
`rustc 1.66.1 (90743e729 2023-01-10)`

### Reproducible example

create a new project with following deps:
```toml
# cargo.toml

[dependencies]
polars = { git = "https://github.com/pola-rs/polars" }
log = { version = "0.4.17", features = ["kv_unstable_serde"] }
```
run `cargo check`:
```bash
# ...many similar errors
error[E0283]: type annotations needed
   --> /home/lictex_/.cargo/git/checkouts/polars-b0d90607192fd414/333a343/polars/polars-core/src/fmt.rs:95:13
    |
95  |             Ok(())
    |             ^^ cannot infer type of the type parameter `E` declared on the enum `Result`
...
114 |                 write_fn(v, $f)?;
    |                                - type must be known at this point
...
304 |                 format_array!(f, self.binary().unwrap(), "binary", self.name(), "Series")
    |                 ------------------------------------------------------------------------- in this macro invocation
    |
    = note: multiple `impl`s satisfying `std::fmt::Error: From<_>` found in the following crates: `core`, `serde_fmt`:
            - impl From<serde_fmt::Error> for std::fmt::Error;
            - impl<T> From<T> for T;
    = note: required for `Result<(), std::fmt::Error>` to implement `FromResidual<Result<Infallible, _>>`
    = note: this error originates in the macro `format_array` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider specifying the generic arguments
    |
95  |             Ok::<(), E>(())
    |               +++++++++

Some errors have detailed explanations: E0282, E0283.
For more information about an error, try `rustc --explain E0282`.
error: could not compile `polars-core` due to 75 previous errors
```

### Expected behavior

no build errors

### Installed versions

<details>
nothing
</details>

--- a/polars/polars-core/src/fmt.rs
+++ b/polars/polars-core/src/fmt.rs
@@ -75,7 +75,7 @@ macro_rules! format_array {
                 .map_or(LIMIT, |n: i64| if n < 0 { $a.len() } else { n as usize });
             std::cmp::min(limit, $a.len())
         };
-        let write_fn = |v, f: &mut Formatter| {
+        let write_fn = |v, f: &mut Formatter| -> fmt::Result {
             if truncate {
                 let v = format!("{}", v);
                 let v_trunc = &v[..v

